src/include/duckdb/execution/operator/csv_scanner/csv_multi_file_info.hpp

```diff
@@ -0,0 +1,64 @@
+
+#pragma once
+
+#include "duckdb/common/multi_file_reader_function.hpp"
+#include "duckdb/execution/operator/csv_scanner/csv_reader_options.hpp"
+
+namespace duckdb {
+
+class CSVFileReaderOptions : public BaseFileReaderOptions {
+public:
+	CSVReaderOptions options;
+};
+
+struct CSVMultiFileInfo {
+	static unique_ptr<BaseFileReaderOptions> InitializeOptions(ClientContext &context);
+	static bool ParseCopyOption(ClientContext &context, const string &key, const vector<Value> &values,
+	                            BaseFileReaderOptions &options, vector<string> &expected_names,
+	                            vector<LogicalType> &expected_types);
+	static bool ParseOption(ClientContext &context, const string &key, const Value &val,
+	                        MultiFileReaderOptions &file_options, BaseFileReaderOptions &options);
+	static void FinalizeCopyBind(ClientContext &context, BaseFileReaderOptions &options,
+	                             const vector<string> &expected_names, const vector<LogicalType> &expected_types);
+	static unique_ptr<TableFunctionData> InitializeBindData(MultiFileBindData &multi_file_data,
+	                                                        unique_ptr<BaseFileReaderOptions> options);
+	static void BindReader(ClientContext &context, vector<LogicalType> &return_types, vector<string> &names,
+	                       MultiFileBindData &bind_data);
+	static void FinalizeBindData(MultiFileBindData &multi_file_data);
+	static void GetBindInfo(const TableFunctionData &bind_data, BindInfo &info);
+	static optional_idx MaxThreads(const MultiFileBindData &bind_data_p, const MultiFileGlobalState &global_state,
+	                               FileExpandResult expand_result);
+	static unique_ptr<GlobalTableFunctionState>
+	InitializeGlobalState(ClientContext &context, MultiFileBindData &bind_data, MultiFileGlobalState &global_state);
+	static unique_ptr<LocalTableFunctionState> InitializeLocalState();
+	static shared_ptr<BaseFileReader> CreateReader(ClientContext &context, GlobalTableFunctionState &gstate,
+	                                               BaseUnionData &union_data, const MultiFileBindData &bind_data_p);
+	static shared_ptr<BaseFileReader> CreateReader(ClientContext &context, GlobalTableFunctionState &gstate,
+	                                               const string &filename, idx_t file_idx,
+	                                               const MultiFileBindData &bind_data);
+	static shared_ptr<BaseFileReader> CreateReader(ClientContext &context, const string &filename,
+	                                               CSVReaderOptions &options,
+	                                               const MultiFileReaderOptions &file_options);
+	static shared_ptr<BaseUnionData> GetUnionData(shared_ptr<BaseFileReader> scan_p, idx_t file_idx);
+	static void FinalizeReader(ClientContext &context, BaseFileReader &reader);
+	static bool TryInitializeScan(ClientContext &context, shared_ptr<BaseFileReader> &reader,
```

Comment:
Is there a reason this is a `shared_ptr<...> &`
Would `BaseFileReader &reader` not suffice?
This could be nullptr, or be set by the function - which as far as I can tell is not the goal?

Reply:
The reason the shared pointer is passed is so the function can take ownership of the reader here